internal/cluster/service.go

```diff
@@ -281,13 +282,21 @@ func (s service) DeleteCluster(ctx context.Context, clusterIdentifier Identifier
 		c, err = s.clusters.GetCluster(ctx, clusterIdentifier.ClusterID)
 	}
 
-	if IsNotFoundError(err) {
-		// Already deleted
+	if IsNotFoundError(err) || c.Status == Deleting {
+		// Already deleted or being deleted
```

Comment:
**Note**: I mean we should split the logic here into two by the two conditions and handle them differently, `IsNotFoundError(err)` should be `204 NoContent` and `c.Status == Deleting` case should be `202 Accepted`. What do you think?

Reply:
Agreed, it feels a little hacky but it can be solved by returning `false, nil` if the status is `DELETING`...

Reply:
Yeah it does, but unfortunately anything else seems levels of complexity higher regarding the needed effort, so I will let it fly temporarily, we need to refactor this part into a more clear behavior/logic in the future.

Reply:
Actually I take it back, it's also `false,nil` when the first delete is initiated, and we want to make it idempotent for subsequent deletes, so I think it fits right into the current logic.
(Not saying it couldn't be better/more clear in an ideal scenario, but it feels more than acceptable in the current circumstances.)